objects.go

```diff
@@ -55,6 +55,40 @@ func NewNetworkView(nv NetworkView) *NetworkView {
 	return &res
 }
 
+type UpgradeStatus struct {
+	IBBase `json:"-"`
+	Ref    string `json:"_ref,omitempty"`
+	Type   string `json:"type"`
+}
+
+// SubElementsStatus object representation
+type SubElementsStatus struct {
+	Ref            string `json:"_ref,omitempty"`
+	CurrentVersion string `json:"current_version"`
+	ElementStatus  string `json:"element_status"`
+	Ipv4Address    string `json:"ipv4_address"`
+	Ipv6Address    string `json:"ipv6_address"`
+	StatusValue    string `json:"status_value"`
+	StepsTotal     int    `json:"steps_total"`
+	StepsCompleted int    `json:"steps_completed"`
+	NodeType       string `json:"type"`
+	Member         string `json:member`
+}
+
+type UpgradeStatusResult struct {
```

Comment:
Is seems that for most of the WAPI objects you are adding (such as this one, UpgradeStatus/UpgradeStatusResult), you have a struct for the object for query, and an object for the query result.  Is this really necessary? What's the reasoning behind that?

Reply:
because when im creating those objects to make requests
getting below errors
"{ \"Error\": \"AdmConProtoError: Field is not searchable: subelements_status\", ","time":"2017/09/22 15:52:36.801"}

though im not sending those. Not sure why, hence created correpsnding result structs to unmarshal the data.

Reply:
I don't see why ntp_setting wouldn't work like other fields. Please investigate.

Reply:
other fields are using array of objects and "nil" checks are passing, but in this case it returns an empty struct for which "omitempty" is not working.

Reply:
What is the error?

Do you really need to embed IBBase in the NTPSetting  class?

Reply:
i dont, i'll remove that, error is the same as i mentioned above. `ntp_setting` is getting queried when doing newgrid, hence it says  "Field is not searchable"

Reply:
OK. You need to add debug to print out the actual WAPI request sent by the connector.